server/src-lib/Hasura/GraphQL/Execute/RemoteJoins.hs

```diff
@@ -164,54 +159,48 @@ emptyResp = GQRespValue OJ.empty VB.empty
 -- | Insert at path, index the value in the larger structure.
```

Comment:
We should make sure that `insertBatchResults` is not slower than O(n) algorithm to begin with, where `n` is number of remote results. This can just be added as a comment and then improved in the later PR.